R/sim_linelist.R

```diff
@@ -20,34 +20,90 @@
 #' * `proportion`: a column with the proportion of the population that are in
 #' that age group. Proportions must sum to one.
 #'
-#' @param contact_distribution An `<epidist>` object or anonymous function for
-#' the contact distribution. This is any discrete density function that
-#' produces non-negative integers (including zero, \eqn{\mathbb{N}_0}) for the
-#' number of contacts per infection.
-#' @param infectious_period An `<epidist>` object or anonymous function for
-#' the infectious period. This defines the duration from becoming infectious
-#' to no longer infectious. In the simulation, individuals are assumed to
+#' @param contact_distribution A `function` or an `<epidist>` object to generate
+#' the number of contacts per infection.
+#'
+#' The function can be defined or anonymous. The function must return a vector
+#' of `numerics`. The vector is the probability mass of sampling a certain
+#' number of contacts, i.e. the first element in the vector is the weight of
+#' sampling zero contacts, the second element is the weight of sampling a single
+#' contact, etc. The function must have a single argument.
+#'
+#' An `<epidist>` can be provided. This will be converted into a probability
+#' mass function internally.
+#'
+#' The default is an anonymous function with a Poisson probability mass function
+#' ([dpois()]) with a mean (\eqn{\lambda}) of 2 contacts per infection.
+#'
+#' @param infectious_period A `function` or an `<epidist>` object for the
+#' infectious period. This defines the duration from becoming infectious to
+#' no longer infectious. In the simulation, individuals are assumed to
 #' become infectious immediately after being infected (the latency period is
 #' assumed to be zero). The time intervals between an infected individual and
 #' their contacts are assumed to be uniformly distributed within the
 #' infectious period. Infectious periods must be strictly positive.
+#'
+#' The function can be defined or anonymous. The function must return a vector
```

Comment:
again, seems like this should be a function returning the probability density of the (real-valued) given number of days?

Reply:
The `infectious_period` should be a function that generates the infectious period for each case in the simulation. The `function` supplied to `infectious_period` will be call directly in the simulation (`.sim_network_bp()`) to generate/sample from the supplied distribution.

> density of the (real-valued) given number of days?

I'm not sure I understand what this means, could you please clarify?

Reply:
will suggest clarification below.